File: Python/Keyboard Row.py
class Solution(object):
    def findWords(self, words):
        """
        :type words: List[str]
        :rtype: List[str]
        """
        self.line1 = set([i for i in 'qwertyuiop'])
        self.line2 = set([i for i in 'asdfghjkl'])
        self.line3 = set([i for i in 'zxcvbnm'])
        return [i for i in words if self.in_one_line(i)]
    
    def in_one_line(self, word):
        word_set = set([i for i in word.lower()])
        if len(word_set | self.line1) == len(self.line1):
            return True
        elif len(word_set | self.line2) == len(self.line2):
            return True
        elif len(word_set | self.line3) == len(self.line3):
            return True
        else:
            return False
        
        

File: cpp/Keyboard Row.cpp
class Solution {
public:
    bool word_in_line(string word, string line) {
        for(char letter : word) {
            letter = letter >= 'A' and letter <= 'Z' ? letter + ('a' - 'A'): letter;
            if (line.find(letter) == string::npos) {
                return false;
            }
        }
        return true;
    }

    vector<string> findWords(vector<string>& words) {
        string line_1 = string("qwertyuiop");
        string line_2 = string("asdfghjkl");
        string line_3 = string("zxcvbnm");
        vector<string> result;
        for(string word : words) {
            if (this->word_in_line(word, line_1) || this->word_in_line(word, line_2) || this->word_in_line(word, line_3)) {
                result.push_back(word);
            }
        }
        return result;
    }
};
